src/components/graph/Code.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react';
+import FileSaver from 'file-saver';
+
+import './Code.css';
+
+class Code extends Component {
+
+  constructor(props){
+    super(props)
+    this.onDownload = this.onDownload.bind(this)
+  }
+
+  onDownload(){
+    let blob = new Blob([this.refs.codeInput.value], {
+       type: "text/plain;charset=utf-8"
```

Comment:
Is it `text/plain` or should we use `text/json`?  It PROBABLY doesn't matter either way. Just asking if there was a reason you went with one over the other.

Reply:
It didn't occur to me to be more specific about the mime type, and this gave me the behavior I was looking for.  I think technically the correct type is ```application/json```, right?   I wonder if browsers might do something annoying, like launch an IDE instead of just downloading the file, if I changed it to that.

Reply:
It probably is `application/json`. If the browser does something funky that's because the user set it up that way? Idk, I always try to be accurate about mime types because I get VERY angry when I open something that is clearly a type of file and it gets displayed as a text file.